### PR TITLE
Updating Document Translation test resources and tests

### PR DESCRIPTION
**Pull request by hamshavathimunibyraiah**
Updating test resource creation: enabling Managed identity on the Translator resource and adding role "Storage Blob Data Contributor" to the Translator Resource on the Storage Account.
Updating DT tests to use MI to access storage accounts. 
Adding tests.yml for Live test pipelines creations.

### CODE REVIEW

File: sdk/translation/azure-ai-translation-document/src/test/java/com/azure/ai/translation/document/DocumentTranslationClientTestBase.java [hunk 1]
```diff
@@ -107,16 +105,12 @@ String getStorageName() {
...
-            : Configuration.getGlobalConfiguration().get("DOCUMENT_TRANSLATION_CONNECTION_STRING");
-    }
-
     BlobContainerClient getBlobContainerClient(String containerName) {
+        String endpoint = String.format("https://%s.blob.core.windows.net", getStorageName());
         BlobContainerClientBuilder blobContainerClientBuilder = new BlobContainerClientBuilder()
+            .endpoint(endpoint)
             .containerName(containerName)
-            .connectionString(getConnectionString());
+            .credential(new DefaultAzureCredentialBuilder().build());
```
Comment (g2vinay) [inline sdk/translation/azure-ai-translation-document/src/test/java/com/azure/ai/translation/document/DocumentTranslationClientTestBase.java:113]: Replace this with below logic: 

```java
.credential(getIdentityTestCredential());
```
Logic for above method.
```java
    public TokenCredential getIdentityTestCredential(InterceptorManager interceptorManager) {
        if (interceptorManager.isPlaybackMode()) {
            return new MockTokenCredential();
        }

        Configuration config = Configuration.getGlobalConfiguration();

        ChainedTokenCredentialBuilder builder = new ChainedTokenCredentialBuilder()
                .addLast(new EnvironmentCredentialBuilder().build())
                .addLast(new AzureCliCredentialBuilder().build())
                .addLast(new AzureDeveloperCliCredentialBuilder().build());


        String serviceConnectionId = config.get("AZURESUBSCRIPTION_SERVICE_CONNECTION_ID");
        String clientId = config.get("AZURESUBSCRIPTION_CLIENT_ID");
        String tenantId = config.get("AZURESUBSCRIPTION_TENANT_ID");
        String systemAccessToken = config.get("SYSTEM_ACCESSTOKEN");

        if (!CoreUtils.isNullOrEmpty(serviceConnectionId)
                && !CoreUtils.isNullOrEmpty(clientId)
                && !CoreUtils.isNullOrEmpty(tenantId)
                && !CoreUtils.isNullOrEmpty(systemAccessToken)) {

            AzurePipelinesCredential azurePipelinesCredential = new AzurePipelinesCredentialBuilder()
                    .systemAccessToken(systemAccessToken)
                    .clientId(clientId)
                    .tenantId(tenantId)
                    .serviceConnectionId(serviceConnectionId)
                    .build();

            builder.addLast(trc -> azurePipelinesCredential.getToken(trc).subscribeOn(Schedulers.boundedElastic()));
        }

        builder.addLast(new AzurePowerShellCredentialBuilder().build());


        return builder.build();
    }
```

Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/src/test/java/com/azure/ai/translation/document/DocumentTranslationClientTestBase.java]: Thank you. Updated

File: sdk/translation/azure-ai-translation-text/src/test/java/com/azure/ai/translation/text/BreakSentenceTests.java [hunk 2]
```diff
@@ -17,7 +18,8 @@ public class BreakSentenceTests extends TextTranslationClientBase {
     public void breakSentenceWithAutoDetect() {
         BreakSentenceItem response = getTranslationClient().findSentenceBoundaries("hello world");
         assertEquals("en", response.getDetectedLanguage().getLanguage());
-        assertEquals(0.98, response.getDetectedLanguage().getConfidence());
+        assertTrue(response.getDetectedLanguage().getConfidence() > 0.8); //Created task: https://machinetranslation.visualstudio.com/MachineTranslation/_workitems/edit/164493
```
Comment (MikeyMCZ) [inline sdk/translation/azure-ai-translation-text/src/test/java/com/azure/ai/translation/text/BreakSentenceTests.java:21]:
```suggestion
        assertTrue(response.getDetectedLanguage().getConfidence() > 0.8);
```

File: sdk/translation/azure-ai-translation-text/src/test/java/com/azure/ai/translation/text/TranslateTests.java [hunk 3]
```diff
@@ -202,7 +203,7 @@ public void translateWithProfanity() {
         assertEquals(1, response.getTranslations().size());
         assertEquals("en", response.getDetectedLanguage().getLanguage());
         assertEquals(1, response.getDetectedLanguage().getConfidence());
-        assertTrue(response.getTranslations().get(0).getText().contains("***"));
+        // assertTrue(response.getTranslations().get(0).getText().contains("***")); Created task: https://machinetranslation.visualstudio.com/MachineTranslation/_workitems/edit/164493
```
Comment (MikeyMCZ) [inline sdk/translation/azure-ai-translation-text/src/test/java/com/azure/ai/translation/text/TranslateTests.java:206]: You can use the source text `shit this is fucking crazy shit fuck`.

**Review by MikeyMCZ — APPROVED** [review]

File: sdk/translation/azure-ai-translation-text/src/test/java/com/azure/ai/translation/text/BreakSentenceTests.java (hunk 2 above)
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-text/src/test/java/com/azure/ai/translation/text/BreakSentenceTests.java]: Updated, here are the live test pipeline results: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4098377&view=ms.vss-test-web.build-test-results-tab

File: sdk/translation/azure-ai-translation-text/src/test/java/com/azure/ai/translation/text/TranslateTests.java (hunk 3 above)
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-text/src/test/java/com/azure/ai/translation/text/TranslateTests.java]: Updated, here are the live test pipeline results: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4098377&view=ms.vss-test-web.build-test-results-tab

**Review by MikeyMCZ — APPROVED** [review]

**Comment by joshfree** [comment]
@g2vinay @billwert could you please do final signoff

**Review by joshfree — APPROVED** [review]

**Pull request merged**